src/sentry/integrations/api/endpoints/data_forwarding_index.py

```diff
@@ -0,0 +1,97 @@
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.apidocs.constants import RESPONSE_BAD_REQUEST, RESPONSE_FORBIDDEN
+from sentry.apidocs.parameters import GlobalParams
+from sentry.integrations.api.serializers.models.data_forwarder import (
+    DataForwarderSerializer as DataForwarderModelSerializer,
+)
+from sentry.integrations.api.serializers.rest_framework.data_forwarder import (
+    DataForwarderSerializer,
+)
+from sentry.integrations.models.data_forwarder import DataForwarder
+from sentry.web.decorators import set_referrer_policy
+
+
+class OrganizationDataForwardingDetailsPermission(OrganizationPermission):
+    scope_map = {
+        "GET": ["org:read"],
+        "POST": ["org:write"],
+    }
+
+
+@region_silo_endpoint
+@extend_schema(tags=["Integrations"])
+class DataForwardingIndexEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.INTEGRATIONS
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    permission_classes = (OrganizationDataForwardingDetailsPermission,)
+
+    @extend_schema(
+        operation_id="Retrieve Data Forwarding Configurations for an Organization",
+        parameters=[GlobalParams.ORG_ID_OR_SLUG],
+        responses={
+            200: DataForwarderModelSerializer,
+        },
+    )
+    @set_referrer_policy("strict-origin-when-cross-origin")
+    @method_decorator(never_cache)
+    def get(self, request: Request, organization) -> Response:
+        queryset = DataForwarder.objects.filter(organization_id=organization.id)
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            on_results=lambda x: serialize(x, request.user),
+            paginator_cls=OffsetPaginator,
+        )
+
+    @extend_schema(
+        operation_id="Create a Data Forwarding Configuration for an Organization",
+        parameters=[GlobalParams.ORG_ID_OR_SLUG],
+        request=DataForwarderSerializer,
+        responses={
+            201: DataForwarderModelSerializer,
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+        },
+    )
+    @set_referrer_policy("strict-origin-when-cross-origin")
+    @method_decorator(never_cache)
+    def post(self, request: Request, organization) -> Response:
+        data = request.data.copy()
+        data["organization_id"] = organization.id
+
+        serializer = DataForwarderSerializer(data=data)
+        if serializer.is_valid():
+            try:
+                with transaction.atomic(using=router.db_for_write(DataForwarder)):
+                    data_forwarder = serializer.save()
```

Comment:
We also might be able to just skip this transaction altogether, we usually only need this wrapper when we want associated DB queries to only be executed if they all pass without error, but this only has the one `INSERT` so the wrapper isn't doing much. 

If it is creating the `DataForwarderProject` rows as well, than ^Josh's point makes the most sense here.